modules/transcription-service-amberscript/src/main/java/org/opencastproject/transcription/amberscript/AmberscriptTranscriptionService.java

```diff
@@ -264,56 +303,78 @@ public void activate(ComponentContext cc) {
       }
     }
-    logger.info("Default metadata field for calculating the amount of speakers is set to '{}'.", speakerMetadataField);
-
+    logger.info("Default number of speakers is set to '{}'.", numberOfSpeakers);
 
-    Option<String> languageFromDublinCoreOpt = OsgiUtil.getOptCfg(cc.getProperties(), LANGUAGE_FROM_DUBLINCORE);
-    if (languageFromDublinCoreOpt.isSome()) {
+    Option<String> speakerFromDublinCoreOpt = OsgiUtil.getOptCfg(cc.getProperties(), SPEAKER_FROM_DUBLINCORE);
+    if (speakerFromDublinCoreOpt.isSome()) {
       try {
-        languageFromDublinCore = Boolean.parseBoolean(languageFromDublinCoreOpt.get());
+        speakerFromDublinCore = Boolean.parseBoolean(speakerFromDublinCoreOpt.get());
       } catch (Exception e) {
-        logger.warn("Configuration value for '{}' is invalid, defaulting to false.", LANGUAGE_FROM_DUBLINCORE);
+        logger.warn("Configuration value for '{}' is invalid, defaulting to true.", SPEAKER_FROM_DUBLINCORE);
       }
     }
-    logger.info("Configuration value for '{}' is set to '{}'.", LANGUAGE_FROM_DUBLINCORE, languageFromDublinCore);
+    logger.info("Configuration value for '{}' is set to '{}'.", SPEAKER_FROM_DUBLINCORE, speakerFromDublinCore);
 
-    amberscriptLangUtil = AmberscriptLangUtil.getInstance();
-    int customMapEntriesCount = 0;
-    Option<String> langCodeMapOpt = OsgiUtil.getOptCfg(cc.getProperties(), LANGUAGE_CODE_MAP);
-    if (langCodeMapOpt.isSome()) {
+    Option<String> speakerMetadataFieldOpt = OsgiUtil.getOptCfg(cc.getProperties(), SPEAKER_METADATA_FIELD);
+    if (speakerMetadataFieldOpt.isSome()) {
       try {
-        String langCodeMapStr = langCodeMapOpt.get();
-        if (langCodeMapStr != null) {
-          for (String mapping : langCodeMapStr.split(",")) {
-            String[] mapEntries = mapping.split(":");
-            amberscriptLangUtil.addCustomMapping(mapEntries[0], mapEntries[1]);
-            customMapEntriesCount += 1;
-          }
-        }
-      } catch (Exception e) {
-        logger.warn("Configuration '{}' is invalid. Using just default mapping.", LANGUAGE_CODE_MAP);
+        speakerMetadataField = SpeakerMetadataField.valueOf(speakerMetadataFieldOpt.get());
+      } catch (IllegalArgumentException e) {
+        logger.warn("Value '{}' is invalid for configuration '{}'. Using default: '{}'.",
+            speakerMetadataFieldOpt.get(), SPEAKER_METADATA_FIELD, speakerMetadataField);
       }
     }
-    logger.info("Language code map was set. Added '{}' additional entries.", customMapEntriesCount);
+    logger.info("Default metadata field for calculating the amount of speakers is set to '{}'.", speakerMetadataField);
+
+    Option<String> transcriptionTypeOpt = OsgiUtil.getOptCfg(cc.getProperties(), TRANSCRIPTIONTYPE);
+    if (transcriptionTypeOpt.isSome()) {
+      transcriptionType = transcriptionTypeOpt.get();
+      logger.info("Default transcription type is set to '{}'.", transcriptionType);
```

Comment:
You've got a list of valid types in the config file, shouldn't the input be checked against that?